### PR TITLE
Skip fetching from server when deleting a named resource

### PR DESCRIPTION
**Pull request by liggitt**
Added a test to ensure `kubectl delete $TYPE $NAME` doesn't first GET the resource

### CODE REVIEW

**Comment by liggitt** [comment]
Fixes #6859 
@smarterclayton PTAL

File: pkg/kubectl/resource/builder.go [hunk 1]
```diff
@@ -60,6 +60,7 @@ type Builder struct {
 
 	singleResourceType bool
 	continueOnError    bool
+	skipFetchingNames  bool
```
Comment (derekwaynecarr) [inline pkg/kubectl/resource/builder.go]: I prefer that we name things not in the negative, makes code easier to read.

So make this fetchNames

Comment (derekwaynecarr) [inline pkg/kubectl/resource/builder.go]: name things not in the negative is just as confusing to read ;-)

**Comment by smarterclayton** [comment]
I have a slight variant I want to discuss tomorrow with you

File: pkg/kubectl/resource/builder.go (hunk 1 above)
Comment (liggitt) [inline pkg/kubectl/resource/builder.go]: I agree, but I really wanted the default state (false) to be the normal case (fetch). I couldn't think of a nonnegative name to accomplish that.

Comment (liggitt) [inline pkg/kubectl/resource/builder.go]: Also, that's not the least confusing comment I've seen, notwithstanding anything to the contrary.

**Comment by liggitt** [comment]
Collapsed logic for lazily loading objects into something controlled by the `requireObject` flag, which default to true. Added a test for deleting by tuple.

**Comment by smarterclayton** [comment]
LGTM unless derek has other comments.

**Comment by smarterclayton** [comment]
Shippable flake due to bad code in master, merging

**Pull request merged**